rskj-core/src/test/java/co/rsk/peg/BridgeUtilsTest.java

```diff
@@ -118,40 +119,111 @@ public void testIsLock() {
         Wallet wallet = new BridgeBtcWallet(btcContext, Collections.singletonList(federation));
         Address federationAddress = federation.getAddress();
         wallet.addWatchedAddress(federationAddress, federation.getCreationTime().toEpochMilli());
+        ActivationConfig.ForBlock actForBlock = mock(ActivationConfig.ForBlock.class);
+        when(actForBlock.isActive(any(ConsensusRule.class))).thenReturn(false);
 
         // Tx sending less than the minimum allowed, not a lock tx
         Coin minimumLockValue = bridgeConstants.getMinimumLockTxValue();
         BtcTransaction tx = new BtcTransaction(params);
         tx.addOutput(minimumLockValue.subtract(Coin.CENT), federationAddress);
         tx.addInput(Sha256Hash.ZERO_HASH, 0, new Script(new byte[]{}));
-        assertFalse(BridgeUtils.isPegInTx(tx, federation, btcContext, bridgeConstants));
+        assertFalse(BridgeUtils.isPegInTxAndValidateMinimum(tx, federation, btcContext, bridgeConstants, actForBlock));
 
         // Tx sending 1 btc to the federation, but also spending from the federation addres, the typical release tx, not a lock tx.
         BtcTransaction tx2 = new BtcTransaction(params);
         tx2.addOutput(Coin.COIN, federationAddress);
         TransactionInput txIn = new TransactionInput(params, tx2, new byte[]{}, new TransactionOutPoint(params, 0, Sha256Hash.ZERO_HASH));
         tx2.addInput(txIn);
         signWithNecessaryKeys(bridgeConstants.getGenesisFederation(), BridgeRegTestConstants.REGTEST_FEDERATION_PRIVATE_KEYS, txIn, tx2, bridgeConstants);
+        assertFalse(BridgeUtils.isPegInTxAndValidateMinimum(tx2, federation, btcContext, bridgeConstants, actForBlock));
+
+        // Tx sending 1 btc to the federation, is a lock tx
+        BtcTransaction tx3 = new BtcTransaction(params);
+        tx3.addOutput(Coin.COIN, federationAddress);
+        tx3.addInput(Sha256Hash.ZERO_HASH, 0, new Script(new byte[]{}));
+        assertTrue(BridgeUtils.isPegInTxAndValidateMinimum(tx3, federation, btcContext, bridgeConstants, actForBlock));
+
+        // Tx sending 50 btc to the federation, is a lock tx
+        BtcTransaction tx4 = new BtcTransaction(params);
+        tx4.addOutput(Coin.FIFTY_COINS, federationAddress);
+        tx4.addInput(Sha256Hash.ZERO_HASH, 0, new Script(new byte[]{}));
+        assertTrue(BridgeUtils.isPegInTxAndValidateMinimum(tx4, federation, btcContext, bridgeConstants, actForBlock));
+    }
+
+
+    @Test
+    public void testIsLock_after_iris() {
```

Comment:
Shouldn't all this combinations be different tests? i.e one for sending below minimum, not pegin txs, etc?